public/js/icinga/behavior/navigation.js

```diff
@@ -6,15 +6,21 @@
 
     Icinga.Behaviors = Icinga.Behaviors || {};
 
+    try {
+        var d3 = require("icinga/icinga-php-thirdparty/mbostock/d3");
+    } catch (e) {
+        console.warn('[Navigation] requires d3.js library');
```

Comment:
If you only *warn* here, make sure the remaining code doesn't break in case `d3` is undefined